packages/integration-tests/src/test-e2e/iam_access_drift.test.ts

```diff
@@ -0,0 +1,348 @@
+        if (listRolePoliciesResponse.PolicyNames) {
+          for (const policyName of listRolePoliciesResponse.PolicyNames) {
+            const getRolePolicyResponse: GetRolePolicyCommandOutput =
+              await iamClient.send(
+                new GetRolePolicyCommand({
+                  RoleName: roleName,
+                  PolicyName: policyName,
+                })
+              );
+            if (getRolePolicyResponse.PolicyDocument) {
+              policies.push({
+                roleName,
+                policyName,
+                policyStatement: decodeURIComponent(
+                  getRolePolicyResponse.PolicyDocument
+                ),
+              });
+            }
+          }
+        }
+      } while (nextToken);
+    }
+    return policies;
+  };
+
+  const reinstallDependencies = async (): Promise<void> => {
+    await fsp.rm(path.join(tempDir, 'node_modules'), {
+      recursive: true,
+      force: true,
+    });
+    await fsp.unlink(path.join(tempDir, 'package-lock.json'));
+
+    await execa('npm', ['install'], {
+      cwd: tempDir,
+      stdio: 'inherit',
+    });
+  };
+
+  const comparePolicy = async (
+    baselinePolicy: string,
+    currentPolicy: string,
+    policyType: AccessCheckPolicyType
+  ): Promise<CheckNoNewAccessCommandOutput> => {
+    return await accessAnalyzerClient.send(
+      new CheckNoNewAccessCommand({
+        existingPolicyDocument: baselinePolicy,
+        newPolicyDocument: currentPolicy,
+        policyType: policyType,
+      })
+    );
+  };
+
+  void it('should not drift iam policies', async () => {
+    await baselineNpmProxyController.setUp();
+
+    await execa('npm', ['create', amplifyAtTag, '--yes'], {
+      cwd: tempDir,
+      stdio: 'inherit',
+    });
+
+    await deploy();
+
+    const baselineManagedIamPolicies = await listManagedIamPolicies();
+    const baselineRoleTrustPolicies = await listRoleTrustPolicies();
+    const baselineRoleInlinePolicies = await listRoleInlinePolicies();
+
+    await baselineNpmProxyController.tearDown();
+    await currentNpmProxyController.setUp();
+    await reinstallDependencies();
+
+    await deploy();
+
+    const currentManagedIamPolicies = await listManagedIamPolicies();
+    const currentRoleTrustPolicies = await listRoleTrustPolicies();
+    const currentRoleInlinePolicies = await listRoleInlinePolicies();
+
+    const comparisonResults: Array<PolicyComparisonResult> = [];
+    for (const baselinePolicy of baselineManagedIamPolicies) {
+      const currentPolicy = currentManagedIamPolicies.find(
+        (p) => p.policyArn === baselinePolicy.policyArn
+      );
+      if (currentPolicy) {
+        comparisonResults.push({
+          baselinePolicy,
+          currentPolicy,
+          comparisonResult: await comparePolicy(
+            baselinePolicy.policyStatement,
+            currentPolicy.policyStatement,
+            'IDENTITY_POLICY'
+          ),
+        });
+      }
```

Comment:
Shouldn't `else` be a fail?

Reply:
It should not. If there's no "current policy" that matches baseline it indicates that policy got removed which is example of "reducing access" (which is fine).

If by any chance policy gets renamed, this will be handled by handling this case here https://github.com/aws-amplify/amplify-backend/pull/1413#discussion_r1592567795

Reply:
on the other hand removal of policy seems like potential breaking change. we can make the check sensitive to that initially and see how this goes.